crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -103,10 +101,7 @@ def adminrepocheck
     else
       render json: check
     end
-  rescue Crowbar::Error::StartStepRunningError,
-         Crowbar::Error::StartStepOrderError,
-         Crowbar::Error::EndStepRunningError,
-         Crowbar::Error::SaveUpgradeStatusError => e
+  rescue Crowbar::Error::UpgradeError => e
```

Comment:
You should catch SystemError here too. It's caught in the model and will set failed status but the response will be still generic 500.